chainer_chemistry/dataset/parsers/csv_file_parser.py

```diff
@@ -107,9 +108,9 @@ def parse(self, filepath, retain_smiles=False):
                     if self.postprocess_label is not None:
                         labels = self.postprocess_label(labels)
 
-                    if retain_smiles:
+                    if return_smiles:
                         assert standardized_smiles == Chem.MolToSmiles(mol)
-                        self.smiles.append(standardized_smiles)
+                        smiles_list.append(standardized_smiles)
```

Comment:
I would like to confirm the motivation for the `return_smiles` option. I thought the typical use case of this option is to turn it off when users want to reduce memory consumption, as we can expect computation time does not decrease so much, as appending to a list does not require take much time and it does create `smiles` anyway even if `return_smiles` is `False`.

Reply:
Most of the time, smiles information is not necessary (All the existing train examples code does not require it).
So even this is small overhead for computation, I thought it nice to omit when possible.
Also, it uses some amount of memory to keep all the standardized smiles.

Reply:
OK, understood.